### ISSUE TITLE
`utils/setup.php` ignores parts of DB configuration in `CONST_Database_DSN`

### ISSUE THREAD

**Issue by paulmenzel** [openstreetmap/Nominatim#318]
Currently, having the PostgreSQL DBMS installed on a different host than localhost, using `utils/setup.php` fails even if configuring it in `settings/local.php` with `CONST_Database_DSN`.

```
@define('CONST_Database_DSN', 'pgsql://nominatim:password@db/nominatim'); // <driver>://<username>:<password>@<host>:<port>/<database>
```

The reason is that all the PostgreSQL commands get not passed the hostname using the switch `--hostname`/`-h`.

A workaround is to use [PostgreSQL’s environment variables](https://www.postgresql.org/docs/9.4/static/libpq-envars.html) `PGHOST`, `PGUSER` and `PGPASSWORD`.

PS: Unfortunately, `utils/setup.php` then still fails when a database function is to be created using a compiled library from Nominatim.

```
Osm2pgsql took 49s overall
Functions
CREATE FUNCTION
ERROR:  could not access file "/nominatim/module/nominatim.so": No such file or directory
ERROR: pgsql returned with error code (3)
pgsql returned with error code (3)
```

**Comment by lonvia** [comment]
Having postgres on anything but localhost is currently not officially supported precisely because the custom nominatim module is needed. If you really want to do that, you need to manually copy the file onto your postgres server first.

**Comment by michael-henderson** [comment]
It would be very handy to be able to spin up an RDS cluster and have nominatim containers talk to it.  For autoscaling this makes a lot of sense.  Are there any plans to work on this one?

**Comment by mtmail** [comment]
You'd need to be able to copy the compiled `nominatim.so` file to the database server. It kind of works if you have control over the server, e.g. docker or kubernetes setups, it still requires custom logic (copying the file) that the `setup.php` or Nominatim installation instructions couldn't cover. With cloud database services like Amazon RDS that's afaik not possible.

**Comment by lonvia** [comment]
Fixed in #1095.
